cpp/CMakeLists.txt

```diff
@@ -34,60 +34,31 @@ if(NOT CMAKE_BUILD_TYPE AND NOT CMAKE_CONFIGURATION_TYPES)
     "Debug" "Release" "MinSizeRel" "RelWithDebInfo")
 endif()
 
-###################################################################################################
-# - compiler options ------------------------------------------------------------------------------
+# Include the nvtx-c directory to access the nvtx-c CMake target
+add_subdirectory(nvtx-c)
 
-set(CMAKE_CXX_STANDARD 11)
-set(CMAKE_CXX_STANDARD_REQUIRED ON)
+###################################################################################################
+# - nvtx target -----------------------------------------------------------------------------------
+add_library(nvtx3-cpp INTERFACE)
+target_include_directories(nvtx3-cpp INTERFACE
+                           "$<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>"
+                           "$<INSTALL_INTERFACE:include>")
 
-if(CMAKE_COMPILER_IS_GNUCXX)
-    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Werror")
-endif(CMAKE_COMPILER_IS_GNUCXX)
+target_compile_features(nvtx3-cpp INTERFACE cxx_std_11)                           
```

Comment:
So I'm curious about this...   Does it mean if your project doesn't otherwise specify a C++ standard that it'll pick C++11?  I assume if you try to set the standard to 98 and use this library, it'll fail.  But is there a way to say "if something in the build limits the standard to C++11, that's ok, but if nothing specifies a standard explicitly, we'd prefer a minimum of C++14?"  It would be nice to get that relaxed-constexpr enabled by default whenever possible.  Especially since GCC has defaulted to C++14 since 6.x, and Microsoft doesn't actually have the ability to specify less than C++14 anymore.

Reply:
This sets the minimum allowed C++ standard version to C++11. I don't know of a way to do what you suggest, but I'll ask people who know more about CMake if it is.

Reply:
[We had a side thread in Slack]  Let's go with C++14 minimum since the C API will work just fine for people who are on old compilers that don't support C++14 properly.  That would be GCC <5, or Visual Studio <2017.